scripts/util.py

```diff
@@ -51,8 +64,74 @@ def read_source_root_path():
 def read_proof_root_path():
     return read_path_from_stdin("the 'proofs' directory (usually '.')")
 
-def read_litani_path():
-    return read_path_from_stdin("the litani script")
+def is_litani_submodule_in_path(path):
+    is_in_path = False
+    try:
+        sys.path.insert(1, path)
+        importlib.import_module("litani")
+        is_in_path = True
+    except ModuleNotFoundError:
+        print(f"Litani submodule was not found at directory: {path})")
+    del sys.path[1]
+    return is_in_path
+
+def add_litani_submodule(path):
+    """Adds the litani submodule to the specified path"""
+    try:
+        subprocess.run(
+             ["git", "submodule", "add", "--quiet", "--name", "litani", "--",
+             "https://github.com/awslabs/aws-build-accumulator.git", "litani"],
+             check=True, cwd=path)
+    except subprocess.CalledProcessError:
+        print(f"Could not add Litani submodule at path: {path}")
+
+def is_litani_installed():
+    return shutil.which("litani") is not None
+
+def determine_litani_origin():
+    choice = None
+    while True:
+        print(LITANI_DIALOG, file=sys.stderr)
+        choice = read_from_stdin()
+        if choice in ("1", "2", "3"):
+            break
+        print("You have entered an invalid value. Please choose again.")
+    return choice
+
+def install_litani_brew_package():
+    """Runs `brew install litani`. Exits if a failure occurs"""
+    try:
+        proc = subprocess.run(
+            ["brew", "install", "litani"],
+            check=False, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+        if proc.returncode:
+            logging.warning(proc.stderr.strip().decode())
+            sys.exit(1)
+    except subprocess.CalledProcessError as e:
+        logging.warning("Failed in running `brew install litani`")
+        logging.warning(e)
+
+def install_litani_submodule(relpath):
+    absolute_path = os.path.abspath(os.path.expanduser(relpath))
+    if not is_litani_submodule_in_path(absolute_path):
+        print(textwrap.fill(
+            "Do you want the Litani submodule to be added to the "
+            "specified path? [Y/n] "), file=sys.stderr)
+        if read_from_stdin().lower() in ("y", "yes"):
+            add_litani_submodule(relpath)
+    return f"$(abspath $(PROOF_ROOT)/{relpath})"
+
+def get_litani():
```

Comment:
I don't like how decoupled this logic is from the actual messages to the user.

Can the text in 🌮 be something like:
```
print("choose an option: ")
for idx, choice in INSTALL_CHOICES.items():
  print(f"{idx}. {choice['description']}")
```
`INSTALL_CHOICES` is
```
{
  "1": {
     "description": "Run `brew install litani`",
     "action": install_litani_brew_package,
  }
```
Then this method is basically
```
return INSTALL_CHOICES[determine_litani_origin()]["action"]()
```

Reply:
This looks much clearer. I can add the `INSTALL_CHOICES` dictionary, however, it will only be a key-value pair of index to description. The reason being that the action for index 2. (i.e. install Litani using the submodule approach) requires a parameter to be passed. This requirement breaks the neat one-liner `return` statement: `return INSTALL_CHOICES[determine_litani_origin()]["action"]()` since I wouldn't be able to pass the `proof_root` argument that is needed by the second action